Extensions/Physics3DBehavior/PhysicsCharacter3DRuntimeBehavior.ts

```diff
@@ -1426,5 +1434,79 @@ namespace gdjs {
         this.updateCharacterPosition();
       }
     }
+
+    export class CharacterCollisionChecker
+      implements gdjs.Physics3DRuntimeBehavior.CollisionChecker {
+      characterBehavior: gdjs.PhysicsCharacter3DRuntimeBehavior;
+
+      _currentContacts: Array<Physics3DRuntimeBehavior> = [];
+      _previousContacts: Array<Physics3DRuntimeBehavior> = [];
+
+      constructor(characterBehavior: gdjs.PhysicsCharacter3DRuntimeBehavior) {
+        this.characterBehavior = characterBehavior;
+      }
+
+      clearContacts(): void {
+        this._previousContacts.length = 0;
+        this._currentContacts.length = 0;
+      }
+
+      updateContacts(): void {
+        const swap = this._previousContacts;
+        this._previousContacts = this._currentContacts;
+        this._currentContacts = swap;
```

Comment:
Because objects/arrays are assigned by reference, I'm almost sure here you erased the previous contact by the same object in memory as the current ones.

Reply:
Actually my bad, the swap works!